scripts/delayed.js

```diff
@@ -1,49 +1,68 @@
 /* eslint-disable import/no-cycle */
-import { getConfigValue } from './configs.js';
+import { FetchGraphQL } from '@dropins/tools/fetch-graphql.js';
+import { getHeaders } from './configs.js';
 import { getUserTokenCookie } from './initializers/index.js';
 import { getConsent } from './scripts.js';
+import { commerceEndpointWithQueryParams } from './commerce.js';
 
 async function initAnalytics() {
-  // Load Commerce events SDK and collector
-  if (getConsent('commerce-collection')) {
-    const config = {
-      environmentId: await getConfigValue('commerce.headers.cs.Magento-Environment-Id'),
-      environment: await getConfigValue('commerce-environment'),
-      storeUrl: await getConfigValue('commerce-store-url'),
-      websiteId: parseInt(await getConfigValue('commerce-website-id'), 10),
-      websiteCode: await getConfigValue('commerce.headers.cs.Magento-Website-Code'),
-      storeId: parseInt(await getConfigValue('commerce-store-id'), 10),
-      storeCode: await getConfigValue('commerce.headers.cs.Magento-Store-Code'),
-      storeViewId: parseInt(await getConfigValue('commerce-store-view-id'), 10),
-      storeViewCode: await getConfigValue('commerce.headers.cs.Magento-Store-View-Code'),
-      websiteName: await getConfigValue('commerce-website-name'),
-      storeName: await getConfigValue('commerce-store-name'),
-      storeViewName: await getConfigValue('commerce-store-view-name'),
-      baseCurrencyCode: await getConfigValue('commerce-base-currency-code'),
-      storeViewCurrencyCode: await getConfigValue('commerce-base-currency-code'),
-      storefrontTemplate: 'EDS',
-    };
+  const gql = new FetchGraphQL(
+    await commerceEndpointWithQueryParams(),
+    await getHeaders('cs'),
+  );
 
-    window.adobeDataLayer.push(
-      { storefrontInstanceContext: config },
-      { eventForwardingContext: { commerce: true, aep: false } },
-      {
-        shopperContext: {
-          shopperId: getUserTokenCookie() ? 'logged-in' : 'guest',
+  try {
+    // fetch data from storeconfig
+    const { data } = await gql.fetchGraphQl(`
+      query DSContextQuery {  
+        config: dataServicesStorefrontInstanceContext {
+          storeId: store_id
+          websiteId: website_id
+          environment
+          storeCode: store_code
+          storeViewId: store_view_id
+          baseCurrencyCode: base_currency_code
+          storeViewCurrencyCode: store_view_currency_code
+          storeViewName: store_view_name
+          storeName: store_name
+          websiteName: website_name
+          storeCode: store_code
+          websiteCode: website_code
+          storeUrl: store_url
+          environmentId: environment_id
+          storeViewCode: store_view_code
+        }
+      }
+    `, { method: 'GET', cache: 'force-cache' });
+
+    // Load Commerce events SDK and collector
+    if (getConsent('commerce-collection')) {
+      const config = {
+        ...data?.config ?? {},
```

Comment:
Perhaps in fallback we can just set as old values, so if query fails (because of ACCS for example) then it'll just check config. Not ideal, but it moves us towards the dynamic query at least?

Reply:
Just check config? I thought the goal was to remove these values from the config.

Reply:
Where else can we put them if not retrievable by query (on ACCS)?